cmd/relay/workflow.go

```diff
@@ -0,0 +1,251 @@
+		},
+	}
+
+	cmd.Flags().StringP("file", "f", "", "Path to relay workflow file.")
+
+	return cmd
+}
+
+func NewReplaceWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "replace [workflow name]",
+		Short: "Replace the yaml definition of a relay workflow",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, cfgerr := config.GetConfig(cmd.Flags())
+
+			if cfgerr != nil {
+				return cfgerr
+			}
+
+			file, ferr := readFile(cmd)
+
+			if ferr != nil {
+				return ferr
+			}
+
+			workflowName, nerr := getWorkflowName(args)
+
+			if nerr != nil {
+				return nerr
+			}
+
+			log := dialog.NewDialog(cfg)
+
+			log.Info(fmt.Sprint("Replacing workflow ", workflowName))
+
+			client := client.NewClient(cfg)
+
+			workflow, werr := client.GetWorkflow(workflowName)
+
+			if werr != nil {
+				return werr
+			}
+
+			revision, rerr := client.CreateRevision(workflowName, file)
+
+			if rerr != nil {
+				return rerr
+			}
+
+			wr := model.NewWorkflowRevision(workflow.Workflow, revision.Revision)
+
+			wr.Output(cfg)
+
+			log.Info(fmt.Sprintf(`Successfully updated workflow %v
+			
+Updated configuration is visible at %v`, workflowName, format.GuiLink(cfg, "/workflow/%v", workflowName)))
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringP("file", "f", "", "Path to relay workflow file.")
+
+	return cmd
+}
+
+func NewDeleteWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete [workflow name]",
+		Short: "Delete a relay workflow",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, cfgerr := config.GetConfig(cmd.Flags())
+
+			if cfgerr != nil {
+				return cfgerr
+			}
+
+			workflowName, nerr := getWorkflowName(args)
+
+			if nerr != nil {
+				return nerr
+			}
+
+			proceed, cerr := util.Confirm("Are you sure you want to delete this workflow?", cfg)
+
+			if cerr != nil {
+				return cerr
+			}
+
+			if !proceed {
+				return nil
+			}
+
+			log := dialog.NewDialog(cfg)
+
+			log.Info("Deleting workflow...")
+
+			client := client.NewClient(cfg)
+
+			_, err := client.DeleteWorkflow(workflowName)
+
+			if err != nil {
+				return err
+			}
+
+			log.Info("Workflow successfully deleted")
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func getWorkflowName(args []string) (string, errors.Error) {
+	if len(args) > 0 {
+		return args[0], nil
+	}
+
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Print("Workflow name: ")
+	namePrompt, err := reader.ReadString('\n')
+
+	if err != nil {
+		return "", errors.NewWorkflowWorkflowNameReadError().WithCause(err)
+	}
+
+	name := strings.TrimSpace(namePrompt)
+
+	if name == "" {
+		return "", errors.NewWorkflowMissingNameError()
+	}
+
+	return strings.TrimSpace(namePrompt), nil
+}
+
+func readFile(cmd *cobra.Command) (string, errors.Error) {
+	filepath, err := cmd.Flags().GetString("file")
+
+	if err != nil {
+		return "", errors.NewWorkflowWorkflowFileReadError().WithCause(err)
+	}
```

Comment:
Hmm I'm not sure this is really a `workflow_file_read_error`--kinda seems like if we can't get a flag out of `cmd.Flags()` then it means the programmer setup the `cmd` object incorrectly, and therefor it should panic?

Reply:
To be honest with you I'm not entirely sure under what conditions `GetString` will error, their docs aren't very clear on that. I can change it to an `unknown_error` which is what we're using for general "developer fucked up" stuff. If the user runs with `--debug` they'll get to see the underlying cause from viper. We should avoid panicking